### DISCUSSION TITLE
setup full text search for Nextcloud on docker

### DISCUSSION THREAD

**Discussion by giscus** [FarisZR/FarisZR.com#5] [Comments]
# setup full text search for Nextcloud on docker

A guick guide on how to setup nextcloud full text search on docker, using elasticsearch and supervisord to do automatic indexing.

https://fariszr.com/en/nextcloud-fulltextsearch-elasticsearch-docker-setup/

**Comment by JMoVS** [comment]
Hi, thank you for writing this article. I was wondering how you made sure that the elasticsearch is not unprotected (unprotected elastic searches are a common occurence of data leakage)

**Comment by FarisZR** [comment]
You are correct, it's not secure.

I made this for mainly for local use, but since elastic search is only exposed to the docker network, without any port forwarding it shouldn't be an issue, unless Nextcloud or the server itself gets hacked of course.

**Comment by Piefje01** [comment]
Hi,

I have Nextcloud AIO but cant get OCR on Images not working.

**Reply by Piefje01** [reply]
workflow_pdf_converter	could not convert /admin/files/Documents/bbn3.docx, reason: []		2023-11-02T12:30:41+0100
workflow_pdf_converter	Can not find office path. Please make sure to configure "preview_libreoffice_path" in your config file.

**Reply by FarisZR** [reply]
This shouldn't be too hard to fix.
you need to add libreoffice to the `command` sub-element for the container in compose.
they don't do this by default because installing libreoffice requires a lot of dependencies.

more details here:
https://github.com/nextcloud/docker/tree/master/.examples

**Reply by Piefje01** [reply]
Hi, i dont understand "add libreoffice command"

**Reply by FarisZR** [reply]
the `command` element in docker compose for Nextcloud and cron.
it may be `command: sh -c "apt update && apt-get install -y --no-install-recommends tesseract-ocr tesseract-ocr-eng tesseract-ocr-$(YOUR_THREE_LETTER_LANGUAGE_CODE) && /entrypoint.sh apache2-foreground"` if you use debian
you just have to add libreoffice to the package list:

```yaml
command: sh -c "apt update && apt-get install -y --no-install-recommends tesseract-ocr tesseract-ocr-eng tesseract-ocr-$(YOUR_THREE_LETTER_LANGUAGE_CODE) libreoffice && /entrypoint.sh apache2-foreground"
```
